internal/storage/storage_test.go

```diff
@@ -371,3 +578,73 @@ func TestS3Backend(t *testing.T) {
 	log.SetOutput(os.Stdout)
 
 }
+
+func TestSftpBackend(t *testing.T) {
+
+	var buf bytes.Buffer
+	log.SetOutput(&buf)
+
+	testConf.Type = sftpType
+	backend, err := NewBackend(testConf)
+	assert.Nil(t, err, "Backend failed")
+
+	assert.Zero(t, buf.Len(), "Got warning when not expected")
+	buf.Reset()
+
+	sftpBack := backend.(*sftpBackend)
+
+	assert.IsType(t, sftpBack, &sftpBackend{}, "Wrong type from NewBackend with sftp")
+
+	var sftpDoesNotExist = "nonexistent/file"
+	var sftpCreatable = os.TempDir() + "/this/file/exists"
+
+	writer, err := sftpBack.NewFileWriter(sftpCreatable)
+	assert.NotNil(t, writer, "Got a nil reader for writer from sftp")
+	assert.Nil(t, err, "sftp NewFileWriter failed when it shouldn't")
+
+	written, err := writer.Write(writeData)
+	assert.Nil(t, err, "Failure when writing to sftp writer")
+	assert.Equal(t, len(writeData), written, "Did not write all writeData")
+	writer.Close()
+
+	reader, err := sftpBack.NewFileReader(sftpCreatable)
+	assert.Nil(t, err, "sftp NewFileReader failed when it should work")
+	assert.NotNil(t, reader, "Got a nil reader for sftp")
+
+	size, err := sftpBack.GetFileSize(sftpCreatable)
+	assert.Nil(t, err, "sftp GetFileSize failed when it should work")
+	assert.NotNil(t, size, "Got a nil size for sftp")
+	assert.Equal(t, int64(len(writeData)), size, "Got an incorrect file size")
+
+	if reader == nil {
```

Comment:
Should be covered by `assert.NotNil(t, reader, "Got a nil reader for sftp")` above?

Reply:
I see the reasoning but `reader` and `size` are different types of nil, so this is not covered above.

Reply:
For clarity, that's not the one I believe makes this redundant - rather https://github.com/neicnordic/sda-pipeline/blob/d820438593c0281fb8fbc63ecaa049e0bf03182a/internal/storage/storage_test.go#L612 is what I think makes https://github.com/neicnordic/sda-pipeline/blob/d820438593c0281fb8fbc63ecaa049e0bf03182a/internal/storage/storage_test.go#L619-L623 a repeat.

Reply:
That served to avoid a panic when a method on the nill `reader` is called a bit below. But it did look like a repeat so I refactored the assertion into a requirement. Good point.